xmppserver/src/test/java/org/jivesoftware/openfire/session/RemoteInitiatingServerDummy.java

```diff
@@ -0,0 +1,555 @@
+                        root.add(Namespace.get("jabber:server"));
+                        final Element error = root.addElement(QName.get("error", "stream", "http://etherx.jabber.org/streams"));
+                        error.addElement(QName.get("undefined-condition", "urn:ietf:params:xml:ns:xmpp-streams"));
+
+                        send(root.asXML().substring(root.asXML().indexOf(">")+1));
+                        throw new InterruptedIOException("Openfire disabled TLS, we require it.");
+                    }
+                    else
+                    {
+                        initiateTLS();
+                        return true;
+                    }
+                default:
+                    throw new IllegalStateException("This implementation does not supported encryption policy: " + encryptionPolicy);
+            }
+        }
+
+        private void initiateTLS() throws IOException {
+            if (doLog) System.out.println("Initiating TLS...");
+            final Document outbound = DocumentHelper.createDocument();
+            final Element startTls = outbound.addElement(QName.get("starttls", "urn:ietf:params:xml:ns:xmpp-tls"));
+            send(startTls.asXML());
+        }
+
+        private void processStartTLSProceed(Element proceed) throws IOException, NoSuchAlgorithmException, KeyManagementException
+        {
+            if (doLog) System.out.println("Received StartTLS proceed.");
+            if (doLog) System.out.println("Replace the socket with one that will do TLS on the next inbound and outbound data");
+
+            final SSLContext sc = SSLContext.getInstance("TLSv1.2");
+
+            TrustManager[] tm = createTrustManagerThatTrustsAll();
+            SecureRandom random = new SecureRandom();
+
+
+            KeyManager[] km = createKeyManager(new KeyPair(null, null), new X509Certificate[0]);
+            if (generatedPKIX != null) {
+                KeyPair keyPair = generatedPKIX.getKeyPair();
+                X509Certificate[] certificateChain = generatedPKIX.getCertificateChain();
+                km = createKeyManager(keyPair, certificateChain);
+            }
+
+            sc.init(km, tm , random);
+            SSLContext.setDefault(sc);
+
+            final SSLSocket sslSocket = (SSLSocket) ((SSLSocketFactory) SSLSocketFactory.getDefault()).createSocket(socket, null, socket.getPort(), true);
+            sslSocket.setSoTimeout((int) SO_TIMEOUT.toMillis());
+            sslSocket.addHandshakeCompletedListener(event -> { if (doLog) System.out.println("SSL handshake completed: " + event); });
+            sslSocket.startHandshake();
+
+            // Just indicate that we would like to authenticate the client but if client
+            // certificates are self-signed or have no certificate chain then we are still
+            // good
+            //sslSocket.setWantClientAuth(true); // DO WE NEED TO BRING THIS INTO OUR LocalOutgoingServerSessionTest MATRIX?
+            phaser.register();
```

Comment:
Is this a question that needs resolving?

Reply:
I'm happy to let this one linger for now. :)